src/utils/graphTraversalUtil.ts

```diff
@@ -552,7 +560,20 @@ export function getExecutionIdsForSelectedNodes(
       const nodeId = String(node.id)
       return parentExecutionId ? `${parentExecutionId}:${nodeId}` : nodeId
     },
-    initialContext: '',
+    initialContext: parentPath,
     expandSubgraphs: true
   })
 }
+
+function findPartialExecutionPathToGraph(
+  target: LGraph,
+  root: LGraph
+): string | undefined {
+  for (const node of root.nodes)
+    if (node.isSubgraphNode()) {
+      if (node.subgraph === target) return `${node.id}`
+      const subpath = findPartialExecutionPathToGraph(target, node.subgraph)
+      if (subpath !== undefined) return node.id + ':' + subpath
+    }
```

Comment:
nit/personal style preference:
```suggestion
  for (const node of root.nodes) {
    if (!node.isSubgraphNode()) continue

    if (node.subgraph === target) return `${node.id}`
    
    const subpath = findPartialExecutionPathToGraph(target, node.subgraph)
    if (subpath !== undefined) return node.id + ':' + subpath
    }
  }
```